ooni/reporter.py

```diff
@@ -257,22 +260,55 @@ def validateCollectorAddress(self):
         if not re.match(regexp, self.collectorAddress):
             raise errors.InvalidOONIBCollectorAddress
 
+    def serializeEntry(self, entry, serialisation_format="yaml"):
+        if serialisation_format == "json":
+            if isinstance(entry, Measurement):
+                report_entry = {
+                    'input': entry.testInstance.report.pop('input', None),
+                    'id': str(uuid.uuid4()),
+                    'test_start_time': entry.testInstance.report.pop('test_start_time', None),
+                    'test_runtime': entry.testInstance.report.pop('test_runtime', None),
+                    'test_keys': entry.testInstance.report
+                }
+            elif isinstance(entry, dict):
+                report_entry = {
+                    'input': entry.pop('input', None),
+                    'id': str(uuid.uuid4()),
+                    'test_start_time': entry.pop('test_start_time', None),
+                    'test_runtime': entry.pop('test_runtime', None),
+                    'test_keys': entry
+                }
+            else:
+                raise Exception("Failed to serialise entry")
+            report_entry.update(self.testDetails)
+            return report_entry
+        else:
+            content = '---\n'
+            if isinstance(entry, Measurement):
+                report_entry = entry.testInstance.report
+            elif isinstance(entry, dict):
+                report_entry = entry
+            else:
+                raise Exception("Failed to serialise entry")
+            content += safe_dump(report_entry)
+            content += '...\n'
+            return content
+
     @defer.inlineCallbacks
     def writeReportEntry(self, entry):
         log.debug("Writing report with OONIB reporter")
-        content = '---\n'
-        if isinstance(entry, Measurement):
-            content += safe_dump(entry.testInstance.report)
-        elif isinstance(entry, Failure):
-            content += entry.value
-        elif isinstance(entry, dict):
-            content += safe_dump(entry)
-        content += '...\n'
 
-        url = self.collectorAddress + '/report'
+        url = self.collectorAddress + '/report/' + self.reportID
```

Comment:
This is because now you "open" the report using `/report/` and then proceed with posting to `/report/$id`, right?